pkg/registry/controller/rest.go

```diff
@@ -42,12 +42,19 @@ func (rcStrategy) NamespaceScoped() bool {
 	return true
 }
 
-// ResetBeforeCreate clears the status of a replication controller before creation.
-func (rcStrategy) ResetBeforeCreate(obj runtime.Object) {
+// PrepareForCreate clears the status of a replication controller before creation.
+func (rcStrategy) PrepareForCreate(obj runtime.Object) {
 	controller := obj.(*api.ReplicationController)
 	controller.Status = api.ReplicationControllerStatus{}
 }
 
+// PrepareForUpdate clears fields that are not allowed to be set by end users on update.
+func (rcStrategy) PrepareForUpdate(obj, old runtime.Object) {
+	newController := obj.(*api.ReplicationController)
+	oldController := old.(*api.ReplicationController)
+	newController.Status = oldController.Status
```

Comment:
Humm, clearly I do not fully comprehend this logic.  The existence of this line causes integration tests to fail.

```
E0325 20:15:57.728446   10801 replication_controller.go:175] unexpected sync error: replicationControllers "nginx-controller" cannot be updated: the resource was updated to 1306
```

Inspecting old and new I see new has Replicas = 2, and old has Replicas = 0.

I am not sure how to track this, and my lack of understanding makes me scared to use it.